nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/AddPermissionTest.java

```diff
@@ -421,6 +426,37 @@ public void shouldAddPermissionForUserWhenBlockingInheritance() throws Operation
         assertExpectedPermissions(acl, params, "existingUser1");
     }
 
+    @Test
+    public void shouldAddPermissionForUsersWhenUsingChainExtension() throws OperationException {
+        // Mock the users params
+        List<String> users = Arrays.asList("jchastain", "rdeniro", "dwashington");
+        users.forEach(user -> when(userManager.getUserModel(user)).thenReturn(new SimpleDocumentModel(user)));
+
+        DocumentModel doc = session.getDocument(new PathRef("/src"));
+        OperationContext ctx = new OperationContext(session);
+        ctx.setInput(doc);
+
+        Object object = automationService.run(ctx, "testAddPermissionToUsers");
+        assertThat(object, instanceOf(DocumentModel.class));
```

Comment:
Not strictly needed, we know that the operation returns a `DocumentModel`, otherwise it would have failed.

Reply:
I take into account the fact that the method returns an Object and in this case i don't have any informations of the real type (Operation service is not generic), so I take the assumption that the type of return can be anything. it's for this reason , i check the type of my Object to avoid any cast exception in my test if for any reason the type is not DocumentModel. what you do think :)

Reply:
Sure!

Reply:
In most Automation calls we are casting directly the object to the one expected, as we know the expected return type of the operation/chain.
So I would go for a simple
```java
DocumentModel documentModel = (DocumentModel) automationService.run(ctx, "testAddPermissionToUsers");
```
and remove the unnecessary `assertThat`.

If we get a `ClassCastException` in the test, that means there is an issue with the chain.

Reply:
Yes the point is that it's not worth using the test framework for _everything_, as long as the test fails it's already a very important datapoint.

Reply:
yep,